invenio_accounts/datastore.py

```diff
@@ -73,11 +73,16 @@ def commit(self):
 
     def mark_changed(self, sid, uid=None, rid=None, model=None):
         """Save a user to the changed history."""
+        # needed so that we have the id from the DB on the model
+        self.db.session.flush()
```

Comment:
@slint Can you have a look at the impact of having a flush here.

Reply:
I see we very often tiptoe around flushes, and get them in implicitly just by chance because we perform another `SELECT` query down the line, e.g.:

```python
domain = Domain.create(...)

# ...in another function...

user = UserAggregate.get(...)  # <-- this triggered a flush, so our `domain`
                               #     from above now has a `domain.id`
```

So I think it's better to be explicit about it, which is what calling `mark_changed` also implies (i.e. that something will be persisted to the DB).

@carlinmack is also checking this change on tests across other repos to see if there are wrong expectations.